### PR TITLE
Package name  bug#1015

### LINKED ISSUE #1015: PACKAGE_NAME_INCORRECT_PATH: incorrectly calculates PATH to the directory if the directory is named with dots
```D:\save\save-core\src\nativeTest\kotlin\org\cqfn\save\core\test.utils\TestUtilsNative.kt:7:9: [PACKAGE_NAME_INCORRECT_PATH] package name does not match the directory hierarchy for this file, the real package name should be: org.cqfn.save.core.test.utils```

This happens because of the dot in "test.utils"

### PR DESCRIPTION
**Pull request by kentr0w**
### What's done:
- Fixed bug
- Added test
- Closes #1015

### CODE REVIEW

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/FileUtils.kt [hunk 1]
```diff
@@ -7,14 +7,15 @@ package org.cqfn.diktat.ruleset.utils
 internal const val SRC_DIRECTORY_NAME = "src"
 
 /**
- * Splits [this] string by file path separator
+ * Splits [this] string by file path separator. Drop last element - file type
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/FileUtils.kt:10]: Please update kdoc correctly. Also, file can have more than one dots. You should drop last after splitting by directory separator and only then split remaining tokens by dot

File: diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter1/PackageNamingWarnTest.kt [hunk 2]
```diff
@@ -369,4 +369,15 @@ class PackageNamingWarnTest : LintTestBase(::PackageNaming) {
...
     }
+
+    @Test
+    @Tag(WarningNames.PACKAGE_NAME_INCORRECT_PATH)
+    fun `shouldn't trigger if path contains dot`() {
+        lintMethod(
+            """
+                |package org.cqfn.diktat.test.utils
+            """.trimMargin(),
+            fileName = "/home/testu/project/src/main/kotlin/org/cqfn/diktat/test.utils/Example.kt",
```
Comment (petertrr) [inline diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter1/PackageNamingWarnTest.kt:380]: `"/home/testu/project/build.gradle.kts"` - will this suggest package name `build.gradle` with your current logic?

**Review by petertrr — CHANGES_REQUESTED** [review]

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/FileUtils.kt (hunk 1 above)
Comment (kentr0w) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/FileUtils.kt]: Fixed

File: diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter1/PackageNamingWarnTest.kt (hunk 2 above)
Comment (kentr0w) [reply diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter1/PackageNamingWarnTest.kt:380]: No, I added test for this case

**Review by petertrr — APPROVED** [review]

**Pull request merged**